libs/blockscout-auth/src/lib.rs

```diff
@@ -1,184 +1,5 @@
-        );
-        let mut headers = HeaderMap::new();
-        headers.insert(COOKIE, cookies.parse().unwrap());
-        headers.insert(CONTENT_TYPE, "application/json".parse().unwrap());
-        if let Some(csrf_token) = csrf_token {
-            headers.insert(
-                HeaderName::from_lowercase(b"x-csrf-token").unwrap(),
-                csrf_token.parse().unwrap(),
-            );
-        };
-        headers
-    }
-
-    fn build_request<T: Serialize>(jwt: &str, csrf_token: Option<&str>, data: T) -> Request<T> {
-        let meta = tonic::metadata::MetadataMap::from_headers(build_headers(jwt, csrf_token));
-        Request::from_parts(meta, Extensions::default(), data)
-    }
-
-    #[test]
-    fn extract_jwt_works() {
-        let jwt = "VALID_JWT_TOKEN";
-        let meta = tonic::metadata::MetadataMap::from_headers(build_headers(jwt, None));
-
-        let token = extract_jwt(&meta).expect("failed to extract metadata");
-        assert_eq!(token, jwt);
-    }
-
-    #[test]
-    fn extract_csrf_token_works() {
-        let csrf_token = "VALUD_CSRF_TOKEN";
-        let meta = tonic::metadata::MetadataMap::from_headers(build_headers("", Some(csrf_token)));
-
-        let token = extract_csrf_token(&meta).expect("failed to extract metadata");
-        assert_eq!(token, csrf_token);
-    }
-
-    #[derive(Debug, Serialize)]
-    struct GetBody {}
-
-    #[derive(Debug, Serialize)]
-    struct PostBody {
-        name: String,
-    }
-
-    #[tokio::test]
-    async fn auth_works() {
-        let jwt = "VALID_JWT_TOKEN";
-        let request = build_request(jwt, None, GetBody {});
-        // TODO: replace with blockscout api mock
-        let success = auth_from_metadata(request.metadata(), true, "")
-            .await
-            .expect("failed to auth");
-        assert_eq!(success.user_id, jwt);
-
-        let jwt = "VALID_JWT_TOKEN";
-        let csrf_token = "_PLUS_CSRF";
-        let request = build_request(
-            jwt,
-            Some(csrf_token),
-            PostBody {
-                name: "lev".to_string(),
-            },
-        );
-        let success = auth_from_metadata(request.metadata(), false, "")
-            .await
-            .expect("failed to auth");
-        assert_eq!(success.user_id, format!("{jwt}{csrf_token}"));
-
-        let request = build_request(
-            jwt,
-            None,
-            PostBody {
-                name: "lev".to_string(),
-            },
-        );
-        auth_from_metadata(request.metadata(), false, "")
-            .await
-            .expect_err("success response from request without csrf_token");
-
-        let request = Request::new(GetBody {});
-        auth_from_metadata(request.metadata(), true, "")
-            .await
-            .expect_err("success response for empty request");
-    }
-}
+pub use auth::{auth_from_metadata, auth_from_tokens, AuthSuccess, Error};
+pub use mock::{init_mocked_blockscout_auth_service, MockUser};
```

Comment:
What making the public is for?

Reply:
I will need to run mock server in admin-rs